iceoryx_posh/source/popo/ports/interface_port_data.cpp

```diff
@@ -22,10 +22,7 @@ namespace iox
 namespace popo
 {
 InterfacePortData::InterfacePortData(const RuntimeName_t& runtimeName, const capro::Interfaces interface) noexcept
-    : BasePortData(capro::ServiceDescription(
-                       capro::InvalidIdString, capro::InvalidIdString, capro::InvalidIdString, {0, 0, 0, 0}, interface),
-                   runtimeName,
-                   "")
+    : BasePortData(capro::ServiceDescription("", "", "", {0, 0, 0, 0}, interface), runtimeName, "")
```

Comment:
A ServiceDescriptionBuilder would be perfect here. Then we could write code like:
```cpp
    : BasePortData(capro::ServiceDescriptionBuilder().interface(interface).create()), runtimeName, "")
```
which looks much cleaner.

This is nothing which has to be done here - but if you want I wont stop you ;)

Reply:
Actually no since in #1036 we remove the creation pattern from the posix wrappers.

But maybe you can implement this here or in a follow up PR. With the helper macro it is also not so much code. Here maybe 10 lines or some more.
Take a look at the MemoryMapBuilder: https://github.com/eclipse-iceoryx/iceoryx/blob/master/iceoryx_hoofs/include/iceoryx_hoofs/internal/posix_wrapper/shared_memory_object/memory_map.hpp

Reply:
Although it's a good idea, I won't do this in this PR. I added a todo to #415 .